src/ILLink.RoslynAnalyzer/TrimAnalysis/TrimAnalysisVisitor.cs

```diff
@@ -297,5 +289,31 @@ public override void HandleReturnValue (MultiValue returnValue, IOperation opera
 				);
 			}
 		}
+
+		static bool TryGetConstantValue (IOperation operation, out MultiValue constValue)
+		{
+			if (operation.ConstantValue.HasValue) {
+				object? constantValue = operation.ConstantValue.Value;
+				if (constantValue == null) {
+					constValue = NullValue.Instance;
+					return true;
+				} else if (operation.Type?.SpecialType == SpecialType.System_String && constantValue is string stringConstantValue) {
+					constValue = new KnownStringValue (stringConstantValue);
+					return true;
+				} else if (operation.Type?.TypeKind == TypeKind.Enum && constantValue is int enumConstantValue) {
+					constValue = new ConstIntValue (enumConstantValue);
+					return true;
+				} else if (operation.Type?.SpecialType == SpecialType.System_Int32 && constantValue is int intConstantValue) {
+					constValue = new ConstIntValue (intConstantValue);
+					return true;
+				} else if (operation.Type?.SpecialType == SpecialType.System_Boolean && constantValue is bool boolConstantValue) {
+					constValue = new ConstIntValue (boolConstantValue ? 1 : 0);
+					return true;
```

Comment:
Why only these types rather than any primitive that can be const?

Reply:
Because only these are recognized by the rest of trimming analysis. And also because we only have value system representation for these (for example `ConstInstValue`... we would need to introduce similar types for other values like `DateTime`, `double`, ...)

Reply:
I'll look into this in a followup change